Cython/Includes/numpy/__init__.pxd

```diff
@@ -34,6 +34,7 @@ cdef extern from *:
 
 cdef extern from "Python.h":
     ctypedef int Py_intptr_t
+    bint PyObject_TypeCheck(object obj, PyTypeObject* type)
```

Comment:
There is a declaration for this in `cpython.object`, which can (and should) just be cimported instead to avoid the duplication here.

Reply:
will update.  any thoughts on if this can/should be nogil?  there was some discussion on this in the numpy thread

Reply:
It's not documented as _not_ requiring the GIL and Pandas apparently doesn't need it according to the discussion over there. We shouldn't take decisions for the CPython project on our side that they may regret at some point.

Reply:
Makes sense.  If/when this goes through, ill make a follow-up PR in numpy to use the cimport for PyObject_TypeCheck